easybuild/framework/easyconfig/tools.py

```diff
@@ -699,3 +701,187 @@ def avail_easyblocks():
                                        easyblock_mod_name, easyblocks[easyblock_mod_name]['loc'], path)
 
     return easyblocks
+
+
+def parse_param_value(string):
+    """Parse specified string as an easyconfig parameter value."""
+
+    def split_one(item, sep=','):
+        """Helper function to split of first part in an item, using given separator."""
+        parts = item.split(sep)
+        return (parts[0], sep.join(parts[1:]))
+
+    param, value = None, None
+
+    # determine list of names of known easyblocks, so we can descriminate an easyblock name
+    easyblock_names = [e['class'] for e in avail_easyblocks().values()]
+
+    # regular expression to recognise a version
+    version_regex = re.compile('^[0-9][0-9.-]')
+
+    # first check whether easyconfig parameter name is specfied as '<param_name>=<value>'
+    if re.match('^[a-z_]+=', string):
+        param, string = split_one(string, sep='=')
+
+    # check if the value is most likely a dictionary '<key>:<val>[;<key>:<val>]'
+    if re.match('^[a-z_]+:', string):
+        value = {}
+        for item in string.split(';'):
+            key, val = split_one(item, sep=':')
+            value[key] = parse_param_value(val)[1]
+
+        # if we encounter a dictionary with (only) 'files' and/or 'dirs' as key(s), it must be sanity_check_paths
+        if len(value) <= 2 and ('files' in value or 'dirs' in value):
+            param = 'sanity_check_paths'
+            for item_key, item_val in value.items():
+                if isinstance(item_val, basestring):
+                    value[item_key] = []
+                    if item_val:
+                        value[item_key].append(item_val)
+                elif isinstance(item_val, tuple):
```

Comment:
No, because `;` is used as a separator between different `<key>:<value>` entries in a dict value. And `,` is the separator for tuples (see below).

`;` is also the separator for lists, but since anything that starts with `[a-z_]+:` is considered a dict value, it'll get split on `;` to separate different `<key>=<value>` entries, and hence you can never have a list-type value anymore... :)

Example: `files:bin/example,lib/libexample.a;dirs:include` gets parsed as:

```python
{
  'files': ['bin/example', 'lib/libexample.a'],
  'dirs': ['include'],
}
```

`;` can not be used as a separator between `bin/example` and `lib/libexample.a`